### PR TITLE
drivers: Use of proper base address macro across all ambiq platforms

### PR DESCRIPTION
**Pull request by RichardSWheatley**
REG_X_BASEADDR will be removed from all hal files.
This forces the use of the peripheral base address

### CODE REVIEW

**Review by AlessandroLuo — DISMISSED** [review]

**Comment by swift-tk** [comment]
We need a private macro to obtain the port in memc_mspi_aps6404l.c and flash_mspi_atxp032.c since these device drivers are meant to be shared. I had clearly missed it in previous PR.
@RichardSWheatley Would you like to fix it?

**Review by gmarull — DISMISSED** [review]
please, make commit message better. Title is misleading.

**Comment by RichardSWheatley** [comment]
> We need a private macro to obtain the port in memc_mspi_aps6404l.c and flash_mspi_atxp032.c since these device drivers are meant to be shared. I had clearly missed it in previous PR. @RichardSWheatley Would you like to fix it?

I can create a ticket and work on it internally, yes.

**Comment by swift-tk** [comment]
> > We need a private macro to obtain the port in memc_mspi_aps6404l.c and flash_mspi_atxp032.c since these device drivers are meant to be shared. I had clearly missed it in previous PR. @RichardSWheatley Would you like to fix it?
> 
> I can create a ticket and work on it internally, yes.

I was hoping you could do it all together in this PR since you are changing the files.

Something like this, private as in platform specific to ambiq but not in hal. Or move the entire section of `CONFIG_SOC_FAMILY_AMBIQ` to mspi_ambiq.h so that it looks cleaner.
image

File: drivers/flash/flash_mspi_atxp032.c [hunk 1]
```diff
@@ -826,7 +826,7 @@ static const struct flash_driver_api flash_mspi_atxp032_api = {
 
 #define FLASH_MSPI_ATXP032(n)                                                                     \
 	static const struct flash_mspi_atxp032_config flash_mspi_atxp032_config_##n = {           \
-		.port = (DT_REG_ADDR(DT_INST_BUS(n)) - REG_MSPI_BASEADDR) /                       \
+		.port = (DT_REG_ADDR(DT_INST_BUS(n)) - MSPI0_BASE) /                       \
```
Comment (swift-tk) [inline drivers/flash/flash_mspi_atxp032.c:829]: Please don't forget to align the slash on the right.

File: drivers/flash/flash_mspi_atxp032.c [hunk 2]
```diff
@@ -826,9 +826,8 @@ static const struct flash_driver_api flash_mspi_atxp032_api = {
 
 #define FLASH_MSPI_ATXP032(n)                                                                     \
 	static const struct flash_mspi_atxp032_config flash_mspi_atxp032_config_##n = {           \
-		.port = (DT_REG_ADDR(DT_INST_BUS(n)) - REG_MSPI_BASEADDR) /                       \
-			(DT_REG_SIZE(DT_INST_BUS(n)) * 4),                                        \
-		.mem_size = DT_INST_PROP(n, size) / 8,                                            \
+		.port        = MSPI_PORT(n),                                                   \
+		.mem_size    = DT_INST_PROP(n, size) / 8,                                      \
```
Comment (swift-tk) [inline drivers/flash/flash_mspi_atxp032.c:830]: The slashes are still misaligned.

File: drivers/mspi/mspi_ambiq.h [hunk 3]
```diff
@@ -47,6 +47,9 @@ enum mspi_ambiq_timing_param {
 	MSPI_AMBIQ_SET_RXDQSDLYEXT    = BIT(7),
 };
 
+#define MSPI_PORT(n)    ((DT_REG_ADDR(DT_INST_BUS(n)) - MSPI0_BASE) /                    \
```
Comment (swift-tk) [inline drivers/mspi/mspi_ambiq.h:50]: Align slash to the 100th column.

**Comment by swift-tk** [comment]
The rest LGTM

**Review by AlessandroLuo — APPROVED** [review]

**Review by swift-tk — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Comment by RichardSWheatley** [comment]
> please, make commit message better. Title is misleading.

@gmarull please review the changes you requested

**Review by gmarull — COMMENTED** [review]
https://github.com/zephyrproject-rtos/zephyr/pull/76707/commits/7079f166fb9b8e02dde1030d3f2bba664d5d9aca Commit title is still unclear/too generic

**Comment by RichardSWheatley** [comment]
> [7079f16](https://github.com/zephyrproject-rtos/zephyr/commit/7079f166fb9b8e02dde1030d3f2bba664d5d9aca) Commit title is still unclear/too generic

It wasn't changed to this by me. It was changed to this by the assignee.

**Comment by gmarull** [comment]
> > [7079f16](https://github.com/zephyrproject-rtos/zephyr/commit/7079f166fb9b8e02dde1030d3f2bba664d5d9aca) Commit title is still unclear/too generic
> 
> It wasn't changed to this by me. It was changed to this by the assignee.

Commit title is still unclear/too generic

**Review by gmarull — DISMISSED** [review]
Please, do not dismiss change requests which are clear and explicit.

**Comment by RichardSWheatley** [comment]
> Please, do not dismiss change requests which are clear and explicit.

They were not and that is why they were dismissed.

"Use of proper base address macro across all ambiq platforms" is the exact and only reason for the change.
If you do not think that is enough, then suggest something else. If not, I will dismiss again.

**Comment by gmarull** [comment]
> > Please, do not dismiss change requests which are clear and explicit.
> 
> They were not and that is why they were dismissed.
> 
> "Use of proper base address macro across all ambiq platforms" is the exact and only reason for the change. If you do not think that is enough, then suggest something else. If not, I will dismiss again.

Isn't this sufficient?
> [7079f16](https://github.com/zephyrproject-rtos/zephyr/commit/7079f166fb9b8e02dde1030d3f2bba664d5d9aca) Commit title is still unclear/too generic

**Comment by RichardSWheatley** [comment]
> > > Please, do not dismiss change requests which are clear and explicit.
> > 
> > 
> > They were not and that is why they were dismissed.
> > "Use of proper base address macro across all ambiq platforms" is the exact and only reason for the change. If you do not think that is enough, then suggest something else. If not, I will dismiss again.
> 
> Isn't this sufficient?
> 
> > [7079f16](https://github.com/zephyrproject-rtos/zephyr/commit/7079f166fb9b8e02dde1030d3f2bba664d5d9aca) Commit title is still unclear/too generic

Yes, my title is sufficient.

**Comment by gmarull** [comment]
> > > > Please, do not dismiss change requests which are clear and explicit.
> > > 
> > > 
> > > They were not and that is why they were dismissed.
> > > "Use of proper base address macro across all ambiq platforms" is the exact and only reason for the change. If you do not think that is enough, then suggest something else. If not, I will dismiss again.
> > 
> > 
> > Isn't this sufficient?
> > > [7079f16](https://github.com/zephyrproject-rtos/zephyr/commit/7079f166fb9b8e02dde1030d3f2bba664d5d9aca) Commit title is still unclear/too generic
> 
> Yes, my title is sufficient.

I guess you're confused about commit and PR titles, which are not the same thing. Finding `drivers: update to use proper Base Address` in git history is just useless, please, make it more clear.

**Comment by RichardSWheatley** [comment]
> drivers: update to use proper Base Address

And yet that is exactly what this PR addresses, so what is your suggestion?

**Comment by RichardSWheatley** [comment]
The video showed me nothing more than [the](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines) which also adds nothing more than the original comment.

What did help was this "are you then updating all drivers in-tree to use "proper" (?) base addresses?"

Had you mentioned that originally, I would have completely understood what you meant in the first place.

**Pull request merged**